src/cpp/security/accesscontrol/Permissions.cpp

```diff
@@ -346,6 +346,46 @@ static bool rfc2253_string_compare(
 }
 
 // Auxiliary functions
+static BIO* load_and_verify_document(
+        X509_STORE* store,
+        BIO* in,
+        SecurityException& exception)
+{
+    BIO* out = nullptr;
+
+    assert(nullptr != store);
+    assert(nullptr != in);
+
+    BIO* indata = nullptr;
+    PKCS7* p7 = SMIME_read_PKCS7(in, &indata);
+    if (nullptr == p7)
+    {
+        exception = _SecurityException_("Cannot read as PKCS7 the file ");
```

Comment:
This log becomes less complete as it is missing the actual file name in the message, but since another more verbose error pops right after when returning `nullptr`, I'd just rephrase this a bit so it does not read as a hanging sentence